src/v/raft/replicate_batcher.cc

```diff
@@ -61,15 +66,15 @@ replicate_stages replicate_batcher::replicate(
 }
 
 ss::future<> replicate_batcher::stop() {
-    // we keep a lock here to make sure that all inflight requests have finished
-    // already
-    return _lock.with([this]() {
-        for (auto& i : _item_cache) {
-            i->_promise.set_exception(
-              std::runtime_error("replicate_batcher destructor called. Cannot "
-                                 "finish replicating pending entries"));
-        }
-        _item_cache.clear();
+    return _bg.close().then([this] {
+        // we keep a lock here to make sure that all inflight requests have
+        // finished already
+        return _lock.with([this] {
```

Comment:
nit: grabbing a mutex here after the gate is closed seems redundant. ostensibly the closed gate would imply there are no other fibers running in this context.